workers/unity/Assets/Fps/Scripts/WorldTiles/MapTemplate.cs

```diff
@@ -0,0 +1,113 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+using Random = System.Random;
+
+namespace Fps
+{
+    [CreateAssetMenu(fileName = "MapTemplate", menuName = "Improbable/Map Template")]
+    public class MapTemplate : ScriptableObject
+    {
+        [SerializeField] internal Texture2D templateBitmap;
+        [SerializeField] internal TileTypeCollection[] tileCollections;
+        [SerializeField] internal TileTypeCollection defaultTileCollection;
+        [SerializeField] internal float unitSize = 1;
+
+        private readonly Dictionary<Color32, TileTypeCollection> tileLookup = new Dictionary<Color32, TileTypeCollection>();
+
+        public GameObject GetTileForLocation(Vector2 location, Random random)
+        {
+            TileTypeCollection collection = defaultTileCollection;
```

Comment:
Why do we copy the variable here?

Reply:
Otherwise a warning about using an unitialized variable usage will be thrown